routes/packages.js

```diff
@@ -13,10 +14,26 @@ router.get('/packages', (req, res, next) => {
 
 // Create a new package for the user
 router.post('/packages', (req, res, next) => {
-  Package.create(req.package).then((item) => {
-    req.user.setPackages(item).then(() => res.json(item));
+  const newPackage = req.body.package;
+  shippoGet(newPackage.carrier, newPackage.trackingNumber).then((response) => {
+    response.json().then((json) => {
+      // If carrier is invalid, shippo will return 404 with body-parser
+      // If carrier is valid but not tracking number, shippo will return null
+      // tracking_status. This is the only known behavior so far...
+      if (!response.ok || !json || !json.tracking_status) {
+        res.status(404).json({ message: 'Could not get package from Shippo' });
```

Comment:
I don't think the status code should be 404 from us. 404 is like the client did a request to us and the resource was not found. A better code is 400 Bad Request to show that the request resource was correct, but there was an issue with the data being bad.